iceoryx_utils/include/iceoryx_utils/internal/concurrent/lockfree_queue/index_queue.hpp

```diff
@@ -95,6 +101,12 @@ class IndexQueue
     /// @return true if an index was obtained, false otherwise
     bool popIfFull(ValueType& index) noexcept;

```

Comment:
Can do, but have to rewrite some client code then.

Reply:
I made them private but since I consider them useful for concise implementations I declared the LockFreeQueue and ResizeableLockFreeQueue as friends. Note that in principle IndexQueue can be used on its own (could replace Loffli), but is also an implementation detail of those other queues. 

By doing so no other clients can access these methods.